ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs

```diff
@@ -41,12 +41,21 @@ public UserSettingsPage(DatabaseContext database) : base(database)
 
         string? avatarHash = await FileHelper.ParseBase64Image(avatar);
 
-        if (avatarHash != null) this.ProfileUser.IconHash = avatarHash;
+        if (avatarHash != null)
+        {
+            // Deny request if in read-only mode
+            if (ServerConfiguration.Instance.UserGeneratedContentLimits.ReadOnlyMode) return this.BadRequest();
```

Comment:
This check should probably be above `FileHelper.ParseBase64Image` since that method will immediately add an asset if it is valid

Reply:
My only concern with this is that I will have to check if the form value `avatar` is not null. Does this value get submitted even if no new avatar is selected?

Reply:
If there's no new avatar then the form value is empty, not sure if that translates to null in the C# controller though. We might be better off just disabling this endpoint in read-only mode unless you're an admin

Reply:
> not sure if that translates to null in the C# controller though.

It should, it is handled by the `ParseBase64Image` method

https://github.com/LBPUnion/ProjectLighthouse/blob/f1c5ad4002ab0a673c7881dcc9c5d1f44f311ddc/ProjectLighthouse/Files/LbpImageHelper.cs#L34